base/iterators.jl

```diff
@@ -1602,4 +1601,81 @@ end
 # be the same as the keys, so this is a valid optimization (see #51631)
 pairs(s::AbstractString) = IterableStatePairs(s)
 
+"""
+    nth(itr, n::Integer)
+
+Get the `n`th element of an iterable collection. Throw a `BoundsError`[@ref] if not existing.
+Will advance any `Stateful`[@ref] iterator.
+
+See also: [`first`](@ref), [`last`](@ref), [`nth`](@ref)
+
+# Examples
+```jldoctest
+julia> Iterators.nth(2:2:10, 4)
+8
+
+julia> Iterators.nth(reshape(1:30, (5,6)), 6)
+6
+
+julia> stateful = Iterators.Stateful(1:10); nth(stateful, 7)
+7
+
+julia> first(stateful)
+8
+```
+"""
+nth(itr, n::Integer) = _nth(itr, n)
+
+# Count
+nth(itr::Count, n::Integer) = n > 0 ? itr.start + itr.step * (n - 1) : throw(ArgumentError("n must be positive."))
+# Repeated
+nth(itr::Repeated, ::Integer) = itr.x
+# Take(Repeated)
+nth(itr::Take{<:Repeated}, n::Integer) =
+    n > itr.n ? throw(BoundsError(itr, n)) : nth(itr.xs, n)
+
+# infinite cycle
+function nth(itr::Cycle{I}, n::Integer) where {I}
+    if IteratorSize(I) isa Union{HasShape, HasLength}
+        _nth(itr.xs, mod1(n, length(itr.xs)))
+    else
+        _nth(itr, n)
+    end
+end
+
+# finite cycle: in reality a Flatten{Take{Repeated{O}}} iterator
+function nth(itr::Flatten{Take{Repeated{O}}}, n::Integer) where {O}
+    if IteratorSize(O) isa Union{HasShape, HasLength}
+        cycles = itr.it.n
+        repeated = itr.it.xs.x
+        k = length(repeated)
+        n > k*cycles ? throw(BoundsError(itr, n)) : _nth(repeated, mod1(n, k))
+    else
+        _nth(itr, n)
+    end
+end
+
+@inline _nth(itr, n) = first(drop(itr, n-1))
```

Comment:
did any benchmarking show these `@inline` are needed? not that these calls shouldn't be inlined, but I'm just curious if there was a situation where the compiler didn't do that already

Reply:
No, I haven't encountered any such scenario. I've put them in as a "just in case". Shall I remove them?